pages/sdk-protocol-kit/reference/safe.md

```diff
@@ -0,0 +1,816 @@
+  safeTxGas, // Optional
+  baseGas, // Optional
+  gasPrice, // Optional
+  gasToken, // Optional
+  refundReceiver, // Optional
+  nonce // Optional
+}
+const safeTransaction = await protocolKit.createEnableFallbackHandlerTx(fallbackHandlerAddress, options)
+```
+
+## Messages
+
+### `createMessage`
+
+Returns a SafeMessage ready to be signed by the owners.
+
+```typescript
+const rayMessage: string | EIP712TypedData = "I am the owner of this Safe"
+const message = protocolKit.createMessage(rawMessage)
+```
+
+### `getSafeMessageHash`
+
+Retrieve the Safe message hash of a string or EIP-712 typed data. It produces the identical hash as invoking the CompatibilityFallbackHandler's getMessageHash method.
+
+```typescript
+const rawMessage = ... // String or EIP-712 typed data
+const messageHash = hashSafeMessage(rawMessage)
+
+const safeMessageHash = await protocolKit.getSafeMessageHash(messageHash)
+```
+
+### `isValidSignature`
+
+Calls the CompatibilityFallbackHandler isValidSignature method (EIP-1271).
+
+It requires 2 parameters:
+
+- messageHash The hash of the message
+- signature The signature to be validated or '0x'. You can send as signature one of the following:
+  1) An array of SafeSignature. In this case the signatures are concatenated for validation (buildSignatureBytes())
+  2) The concatenated signatures as string
+  3) '0x' if you want to validate an onchain message (Approved hash)
+
+The method returns if the signature is valid
+
+```typescript
+const rawMessage = ... // String or EIP-712 typed data
+const messageHash = hashSafeMessage(rawMessage)
+const safeMessageHash = await protocolKit.getSafeMessageHash(messageHash)
+
+const isValidSignature = await protocolKit.isValidSignature(safeMessageHash, signature)
+...
+const isValidSignature = await protocolKit.isValidSignature(safeMessageHash, [signature1, signature2])
+...
+const isValidSignature = await protocolKit.isValidSignature(safeMessageHash, '0x')
+```
+
+### `signMessage`
+
+Returns a new `SafeMessage` object that includes the signature of the current owner. 
+
+You can use multiple signing methods, such as:
+
+- ETH_SIGN (`eth_sign`): Regular hash signature
+- ETH_SIGN_TYPED_DATA_V4 (`eth_signTypedData_v4`): Typed data signature v4, The default method if no signing method is passed
+- ETH_SIGN_TYPED_DATA_V3 `eth_signTypedData_v3`: Typed data signature v3
+- ETH_SIGN_TYPED_DATA `eth_signTypedData`: Typed data signature
+- SAFE_SIGNATURE: Signing with another Safe contract as signer
+
+The third parameter (optional) is the preImageSafeAddress. If the preimage is required, this is the address of the Safe that will be used to calculate the preimage. It's mandatory parameter for 1.3.0 and 1.4.1 contract versions. This is because the safe uses the old EIP-1271 interface which uses `bytes` instead of `bytes32` for the message we need to use the pre-image of the message to calculate the message hash. This parameter is used in conjunction with the SAFE_SIGNATURE signing method.
```

Comment:
```suggestion
The third parameter (optional) is the preImageSafeAddress. If the preimage is required, this is the address of the Safe that will be used to calculate the preimage. It's a mandatory parameter for 1.3.0 and 1.4.1 contract versions. This is because the safe uses the old EIP-1271 interface, which uses `bytes` instead of `bytes32` for the message; we need to use the pre-image of the message to calculate the message hash. This parameter is used in conjunction with the SAFE_SIGNATURE signing method.
```